### PR TITLE
ISPN-14601 SSCAN resp command

### PR DESCRIPTION
**Pull request by rigazilla**
https://issues.redhat.com/browse/ISPN-14601

Based on HSCAN, the content set is turned into a map to respect the iterator framwrk style

### CODE REVIEW

**Comment by tristantarrant** [comment]
@rigazilla please rebase

File: server/resp/src/main/java/org/infinispan/server/resp/commands/set/SSCAN.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.infinispan.server.resp.commands.iteration.BaseIterationCommand;
+
+/**
+ * `<code>HSCAN key cursor [MATCH pattern] [COUNT count]</code>` command.
+ * <p>
+ * Scan the key-value properties of the hash stored under <code>key</code>.
+ *
+ * @since 15.0
+ * @see <a href="https://redis.io/commands/hscan/">Redis Documentation</a>
+ * @author José Bolina
```
Comment (jabolina) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/set/SSCAN.java:26]: :grin: Seems the doc needs an update.

File: server/resp/src/main/java/org/infinispan/server/resp/commands/set/SSCAN.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   protected CompletionStage<IterationInitializationContext> initializeIteration(Resp3Handler handler, List<byte[]> arguments) {
+      EmbeddedSetCache<byte[], byte[]> multimap = handler.getEmbeddedSetCache();
+      return multimap.get(arguments.get(0)).thenApply(entry -> {
+         if (entry == null) return null;
+         return MapIterationInitializationContext.withSource(setToMap(entry));
+      });
+   }
+
+   private Map<byte[],byte[]> setToMap(SetBucket<byte[]> entry) {
+      return entry.toList().stream().collect(Collectors.toMap(v -> v, v-> "0".getBytes()));
```
Comment (jabolina) [inline server/resp/src/main/java/org/infinispan/server/resp/commands/set/SSCAN.java:49]: We could use an empty array in the value. With this approach, I think even filtering would work.

**Review by jabolina — COMMENTED** [review]
I think the approach is good. I am unsure if using the set as a key set would affect something else, so it would be nice to have an additional reviewer.

**Pull request merged**